Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,194 @@
+        } catch {
+            logger.error("Error getting access token", metadata: [
+                "error": "\(error)"
+            ])
+            return .init(statusCode: .badRequest, body: "Error getting access token")
+        }
+
+        let user: User
+
+        do {
+            user = try await getGHUser(accessToken: accessToken)
+        } catch {
+            logger.error("Error getting user ID", metadata: [
+                "error": "\(error)"
+            ])
+            return .init(statusCode: .badRequest, body: "Error getting user")
+        }
+
+        let jwt: GHOAuthPayload
+
+        do {
+            jwt = try await verifyState(state: String(event.queryStringParameters?["state"] ?? ""))
+        } catch {
+            logger.error("Error during state verification", metadata: [
+                "error": "\(error)",
+                "state": .string(event.queryStringParameters?["state"] ?? "")
+            ])
+            return .init(statusCode: .badRequest, body: "Error verifying state")
+        }
+
+        do {
+            try await userService.linkGithubID(discordID: jwt.discordID.rawValue, githubID: "\(user.id)")
+        } catch {
+            logger.error("Error linking user to GitHub account", metadata: [
+                "discordID": .stringConvertible(jwt.discordID),
+                "githubID": .stringConvertible(user.id),
+            ])
+            return .init(statusCode: .badRequest, body: "Error linking user")
+        }
+
+        do {
+            try await discordClient.updateOriginalInteractionResponse(
+                token: jwt.interactionToken,
+                payload: .init(
+                    embeds: [.init(
+                        description: "Successfully linked your Discord account to GitHub account with username: \(user.login)",
+                        color: .green
+                    )]
+                )
+            ).guardSuccess()
+        } catch {
+            logger.warning("Received Discord error while updating interaction", metadata: [
+                "error": "\(error)"
+            ])
+        }
+
+        return .init(statusCode: .ok, body: "Account linking successful, you can return to Discord now")
+    }
+
+    func verifyState(state: String) async throws -> GHOAuthPayload {
+        logger.debug("Retrieving JWT signer secrets to verify state parameter")
+        guard let publicKeyString = ProcessInfo.processInfo.environment["ACCOUNT_LINKING_OAUTH_FLOW_PUB_KEY"] else {
+            throw Errors.envVarNotFound(name: "ACCOUNT_LINKING_OAUTH_FLOW_PUB_KEY")
+        }
+        guard let publicKeyData = Data(base64Encoded: publicKeyString) else {
+            throw Errors.invalidPublicKey
+        }
+        let publicKey = try ECDSAKey.public(pem: publicKeyData)
+
+        let signer = JWTSigner.es256(key: publicKey)
+
+        let payload = try signer.verify(state, as: GHOAuthPayload.self)
+
+        return payload
```

Comment:
```suggestion
        let jwtJsonEncoder = JSONEncoder(); jwtJsonEncoder.dateEncodingStrategy = .integerSecondsSince1970
        let jwtJsonDecoder = JSONDecoder(); jwtJsonDecoder.dateDecodingStrategy = .integerSecondsSince1970
        let signers = JWTSigners(defaultJSONEncoder: jwtJsonEncoder, defaultJSONDecoder: jwtJsonDecoder)
        signers.use(.es256(key: publicKey))
        return try signers.verify(state, as: GHOAuthPayload.self)
```
- Using a `JWTSigner` by itself is (very softly) deprecated in JWTKit. There aren't any warnings or even really documentation to that effect at this time, but it's true nonetheless. Going through `JWTSigners` is always recommended.
- Now that JWTKit's API supports it (which is a brand-new thing), we can actually configure the JSON coders appropriately.

I haven't included it in this suggestion, but the `JWTSigners` object should be created _once_ (including the loading of the key and such) during `init` and saved in a stored property rather than all of that work being done repeatedly here. (Yes I know that's of limited value given the nature of lambdas, but just the same.)